elements/lisk-api-client/src/transaction.ts

```diff
@@ -0,0 +1,178 @@
+		const txInput = input;
+		const networkIdentifier = Buffer.from(this._nodeInfo.networkIdentifier, 'hex');
+		const { publicKey, address } = getAddressAndPublicKeyFromPassphrase(passphrase);
+		const accountHex = await this._channel.invoke<string>('app:getAccount', {
+			address: address.toString('hex'),
+		});
+		const account = decodeAccount(Buffer.from(accountHex, 'hex'), this._schema);
+		if (!txInput.moduleID) {
+			if (!txInput.moduleName) {
+				throw new Error('Missing moduleId and moduleName');
+			} else {
+				const registeredModule = this._nodeInfo.registeredModules.find(
+					module => module.name === input.moduleName,
+				);
+				txInput.moduleID = registeredModule?.id ? registeredModule.id : txInput.moduleID;
+			}
+		}
+		if (txInput.assetID === undefined || null) {
+			if (!txInput.assetName) {
+				throw new Error('Missing assetId and assetName');
+			} else {
+				const registeredAsset = this._nodeInfo.registeredModules.find(
+					asset => asset.name === input.assetName,
+				);
+				txInput.assetID = registeredAsset?.id ? registeredAsset.id : txInput.assetID;
+			}
+		}
+		if (!options?.nonce && !txInput.nonce) {
+			if (
+				typeof account.sequence !== 'object' ||
+				!(account.sequence as Record<string, unknown>).nonce
+			) {
+				throw new Error('Unspported account type.');
+			}
+			txInput.nonce = (account.sequence as { nonce: bigint }).nonce;
+		}
+		if (txInput.senderPublicKey) {
+			txInput.senderPublicKey = publicKey;
+		}
+		const assetSchema = getTransactionAssetSchema(txInput, this._schema);
+		if (account.keys && (account.keys as MultiSignatureKeys).numberOfSignatures > 0) {
+			return signMultiSignatureTransaction(
+				assetSchema,
+				txInput,
+				networkIdentifier,
+				passphrase,
+				account.keys as MultiSignatureKeys,
+				options?.includeSenderSignature,
+			);
+		}
+		if (options?.multisignatureKeys && options?.includeSenderSignature) {
+			return signMultiSignatureTransaction(
+				assetSchema,
+				txInput,
+				networkIdentifier,
+				passphrase,
+				options.multisignatureKeys,
+				options.includeSenderSignature,
+			);
+		}
+		return signTransaction(assetSchema, txInput, networkIdentifier, passphrase);
+	}
+
+	public async get(id: Buffer): Promise<Record<string, unknown>> {
+		const transactionHex = await this._channel.invoke<string>('app:getTransactionByID', {
+			id: id.toString('hex'),
+		});
+		return decodeTransaction(Buffer.from(transactionHex, 'hex'), this._schema);
+	}
+
+	public async getFromPool(): Promise<Record<string, unknown>[]> {
+		return this._channel.invoke('app:getTransactionsFromPool');
+	}
+
+	public sign(
+		transaction: Record<string, unknown>,
+		passphrases: string[],
+	): Record<string, unknown> {
+		const assetSchema = getTransactionAssetSchema(transaction, this._schema);
+		const networkIdentifier = Buffer.from(this._nodeInfo.networkIdentifier, 'hex');
+		return signTransaction(assetSchema, transaction, networkIdentifier, passphrases[0]);
```

Comment:
this should also check if the sender is multi-signature account, or optionally keys are provided, and use `signMultiSignatureTransaction` for each passphrase or `signTransaction` with the first element